client/schnorr_ec.go

```diff
@@ -9,70 +9,110 @@ import (
 	"math/big"
 )
 
-func (c *Client) SchnorrEC(protocolType common.ProtocolType, dlog *dlog.ECDLog, secret *big.Int) error {
-	prover, err := dlogproofs.NewSchnorrECProver(protocolType)
+type SchnorrECClient struct {
+	GenericClient
+	prover  *dlogproofs.SchnorrECProver
+	secret  *big.Int
+	a       *common.ECGroupElement
+	variant pb.SchemaVariant
+}
+
+func NewSchnorrECClient(endpoint string, variant pb.SchemaVariant, dlog *dlog.ECDLog,
+	s *big.Int) (*SchnorrECClient, error) {
+	genericClient, err := NewGenericClient(endpoint)
 	if err != nil {
-		return fmt.Errorf("Could not create schnorr EC prover: %v", err)
+		return nil, err
 	}
 
-	c.handler.schnorrECProver = prover
+	prover, err := dlogproofs.NewSchnorrECProver(common.ToProtocolType(variant))
+	if err != nil {
+		return nil, fmt.Errorf("Could not create schnorr EC prover: %v", err)
+	}
 
-	initMsg := c.getInitialMsg()
+	return &SchnorrECClient{
+		GenericClient: *genericClient,
+		prover:        prover,
+		secret:        s,
+		a: &common.ECGroupElement{
+			X: dlog.Curve.Params().Gx,
+			Y: dlog.Curve.Params().Gy,
+		},
+	}, nil
+}
 
-	a := &common.ECGroupElement{
-		X: dlog.Curve.Params().Gx,
-		Y: dlog.Curve.Params().Gy,
+// Run calls the appropriate function to execute the Schnorr EC protocol.
+func (c *SchnorrECClient) Run() error {
```

Comment:
I'm thinking, would it perhaps be even better to just export `RunSigma()` and `RunZeroKnowledge()` in favor of deleting `Run()`?